### PR TITLE
Document /stats/ JSON API

### PR DESCRIPTION
**Pull request by cooperlees**
- Add new .rst documenting how the `/stats/` JSON API can be queried

Finishes https://github.com/pypa/warehouse/issues/4288

### CODE REVIEW

**Comment by ewdurbin** [comment]
ready to 🚢 when CI passes :)

File: docs/api-reference/stats.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
+STATS API
```
Comment (di) [inline docs/api-reference/stats.rst]:
```suggestion
Stats API
```

File: docs/api-reference/stats.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+=========
+
+PyPI offers a JSON stats endpoint via a `Content-Type: application/json` GET
```
Comment (di) [inline docs/api-reference/stats.rst]:
```suggestion
PyPI offers a JSON stats endpoint via a ``Content-Type: application/json`` GET
```

File: docs/api-reference/stats.rst [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+request. Without the Content-Type header set, you will be returned a HTML page.
```
Comment (di) [inline docs/api-reference/stats.rst]:
```suggestion
request. Without the ``Content-Type`` header set, you will be returned a HTML page.
```

File: docs/api-reference/stats.rst [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        "Fiona": {
+          "size": 6209535709
+        },
+        "FlexGet": {
+          "size": 4387002448
+        },
+        ...
+      },
+      "total_packages_size": 23965450269
+    }
```
Comment (di) [inline docs/api-reference/stats.rst]: This whole block needs indented.

File: docs/api-reference/stats.rst [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    .. code:: http
+
+        GET /stats HTTP/1.1
+        Host: pypi.org
+        Accept: application/json
+        Content-Type: application/json
+
+    **Example response**:
+
+    .. code:: http
```
Comment (di) [inline docs/api-reference/stats.rst]:
```suggestion
    .. code:: json
```

File: docs/api-reference/stats.rst (hunk 1 above)
Comment (cooperlees) [reply docs/api-reference/stats.rst]: JSON API uses all caps so I did. But will change.

Comment (di) [reply docs/api-reference/stats.rst]: Right, because JSON is an acronym -- "Stats" is not (see also the "Legacy API")

**Comment by cooperlees** [comment]
Anything else required from me here guys?

File: docs/api-reference/stats.rst [hunk 6]
```diff
@@ -0,0 +1,49 @@
+Stats API
+=========
+
+PyPI offers a JSON stats endpoint via a ```Content-Type: application/json`` GET
```
Comment (di) [inline docs/api-reference/stats.rst]:
```suggestion
PyPI offers a JSON stats endpoint via a ``Content-Type: application/json`` GET
```

One too many backticks here now.

**Pull request merged**